### ISSUE TITLE
cleanOnValidationError and failed migration

### ISSUE THREAD

**Issue by hohwille** [flyway/flyway#1419]
###### What version of Flyway are you using?

4.0.3
###### Which client are you using? (Command-line, Java API, Maven plugin, Gradle plugin, SBT plugin, ANT tasks)

spring-boot
###### What database are you using (type & version)?

h2, oracle
###### What operating system are you using?

ubuntu linux, windows7
###### What did you do?

(Please include the content causing the issue, any relevant configuration settings, and the command you ran)
1. messed a flyway SQL script
2. run with `cleanOnValidationError`
3. fixed messed SQL
4. rerun with `cleanOnValidationError`
###### What did you expect to see?

DB gets cleaned as the checksum of the failed migration has changed and `cleanOnValidationError` is true.
###### What did you see instead?

Schema "..." contains a failed migration to version .... !

To summarize: When I am working in local development environment, I enable `cleanOnValidationError`. In such case I do not want to manually repair a failed migration. I want to fix the SQL and restart what `cleanOnValidationError` is all about. Of course I never want a failed migration to trigger a clean in production but there I would never ever activate `cleanOnValidationError` and instead would enabled `cleanDisabled`. So IMHO `cleanOnValidationError` should clean if a checksum changed no matter if a migration failed or not. If I am using that feature I say "hey, if something goes wrong, clean DB and restart from scratch!" so why should I care about a failed migration then?

**Comment by axelfontaine** [comment]
I am the feeling this may be a Spring Boot issue. Can you try debugging this to see what's going on?

**Comment by axelfontaine** [comment]
Closing due to lack of response from original reporter
